File: backend/app/schemas/admin_registered_users.py
"""`GET /admin/registered-users` — see docs/API_CONTRACTS.md "GET
/admin/registered-users". Own module (not folded into `admin_stats.py` or
`admin_overview.py`) since this combines two different data sources
(Cognito group membership + the local `user_profile.last_seen_at`) into a
row shape neither of those existing schemas models.
"""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class RegisteredUserOut(BaseModel):
    cognito_sub: str
    email: str | None
    # Cognito `UserStatus` verbatim (e.g. "CONFIRMED", "UNCONFIRMED",
    # "FORCE_CHANGE_PASSWORD", "ARCHIVED", "COMPROMISED", "RESET_REQUIRED")
    # — never remapped/renamed here, so an admin reading this list sees the
    # same vocabulary the Cognito console itself uses.
    status: str
    signup_at: datetime | None
    # `null` = no local `user_profile` row has ever recorded activity for
    # this user yet (never returned since last_seen_at tracking shipped,
    # or returned only before it shipped) — rendered as "Never" by the
    # frontend, not the same as a `0`/epoch value.
    last_seen_at: datetime | None


class RegisteredUsersResponse(BaseModel):
    results: list[RegisteredUserOut]
    page: int
    page_size: int
    total: int

File: frontend/src/types/adminRegisteredUsers.ts
// Types for GET /admin/registered-users, matching docs/API_CONTRACTS.md
// "GET /admin/registered-users" -- the admin "Registered users" report
// (email, Cognito status, signup date, last-visited).

export interface RegisteredUserRow {
  cognito_sub: string;
  email: string | null;
  /** Cognito UserStatus verbatim (CONFIRMED, UNCONFIRMED, ...) -- not remapped. */
  status: string;
  signup_at: string | null;
  /** `null` means never tracked yet -- render as "Never", not a fake date. */
  last_seen_at: string | null;
}

export interface RegisteredUsersResponse {
  results: RegisteredUserRow[];
  page: number;
  page_size: number;
  total: number;
}
